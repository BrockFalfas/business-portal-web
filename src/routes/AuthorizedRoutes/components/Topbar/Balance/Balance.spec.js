import React from 'react';
import { shallow } from 'enzyme';

import { Balance } from './index';

const initBalance = overrides => {
  const mockProps = {
    balanceValue: 0,
    paymentDaysLeft: 0,
    isAutoRenewOn: true,
    onChange: jest.fn(),
  };
  const wrapper = shallow(<Balance {...mockProps} {...overrides} />);
  return { mockProps, wrapper };
};

describe('component: <Balance/>', () => {
  describe('renderBalance', () => {
    it('should render without crashes', () => {
      const { wrapper } = initBalance();
      expect(wrapper).toBeTruthy();
    });

    it('should render as expected', () => {
      const { wrapper } = initBalance({
        balanceValue: 5000,
        paymentDaysLeft: 5,
      });
      expect(wrapper).toMatchSnapshot();
    });

    it('should display autorenew ON or OFF depend flag isAutoRenewOn', () => {
      const { wrapper } = initBalance();

      wrapper.setProps({ isAutoRenewOn: true });
      expect(wrapper.find('.Balance-approximately-switch-label').text()).toBe('Auto Renew ON');

      wrapper.setProps({ isAutoRenewOn: false });
      expect(wrapper.find('.Balance-approximately-switch-label').text()).toBe('Auto Renew OFF');
    });
  });
});
