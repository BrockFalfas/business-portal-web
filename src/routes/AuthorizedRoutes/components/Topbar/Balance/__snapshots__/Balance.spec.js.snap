// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <Balance/> renderBalance should render as expected 1`] = `
<div
  className="Balance"
>
  <div
    className="Balance-value"
  >
    <div
      className="Balance-value-currency"
    >
      Balance:
       
      <img
        alt="Thor Logo grey"
        className="Balance-image"
        src="images/Thor_Logo-small.png"
      />
    </div>
    <span>
      5,000 THOR
    </span>
  </div>
  <div
    className="Balance-approximately-box"
  >
    <div
      className="Balance-approximately-message"
    >
      Approx. 
      5
       days of payments
    </div>
    <div
      className="Balance-approximately-switch"
    >
      <span
        className="Balance-approximately-switch-label"
      >
        Auto Renew 
        ON
      </span>
      <Switch
        defaultChecked={true}
        onChange={[Function]}
      />
    </div>
  </div>
</div>
`;
